.travis.yml

```diff
@@ -97,6 +97,19 @@ jobs:
         - sudo journalctl -u snapd
         - /snap/bin/http https://api.snapcraft.io/v2/snaps/info/core architecture==amd64 Snap-Device-Series:16
 
+    - stage: deploy
```

Comment:
Have you considered using the Travis dpl? Something like:
```
deploy:
  provider: snap
  snap: snapcraft-*.snap
  channel: edge-pr-$TRAVIS_PULL_REQUEST
```

The main question I have is whether or not channel supports variables...

Reply:
Travis dpl from what I saw does not support PRs.